packages/block-library/src/social-links/block.json

```diff
@@ -3,6 +3,10 @@
 	"name": "core/social-links",
 	"category": "widgets",
 	"attributes": {
+		"allowedBlocks": {
+			"type": "array",
+			"default": [ "core/social-link" ]
+		},
```

Comment:
I was wondering if this might be redundant since the block `core/social-link` will be always allowed because [its parent is this block](https://github.com/WordPress/gutenberg/blob/67de26f44e93e6274b822bc79e4b36dbe2588576/packages/block-library/src/social-link/block.json#L5). In fact, I tried to set the `allowedBlocks` to an empty array and I could still see all the social link variants displayed in the inserter.

Reply:
Good catch. I completely forgot about it. It would be great to investigate how those two features interact with each other.

Reply:
So I tried fiddling with the `parent` property, and so far I am able to insert a custom block from a third-party plugin just by setting it a `parent` attribute.

Since child blocks are only available through their parent blocks, we could simply register two blocks that uses the same components for a block to be both available as an inner block and as a 'standalone' block.

For the nesting level concern, may it be something we want? Let's say third-plugin A adds its `a/social-share-link` in `core/social-link` inner blocks, while third-party plugin B adds its own `b/social-profile-avatar-display` which accepts an inner `core/social-link` block, and retrieves the profile picture from the passed social link (throughout some CORS bypass black magic). 

Wouldn't we, in this case, want to prevent the `a/social-share-link` to be available as an inner block of `b/social-profile-avatar-display`?

Reply:
>The biggest limitation of parent is that it won't work with more than one level of nesting.

Yeah, good point. Are the allowed blocks list passed through `allowedBlocks` prop propagated to nested children? I haven't checked this further but it looks like it doesn't neither, right?

Reply:
This was a great catch because I was testing Navigation with the same update and trying to limit the list of blocks and hadn't figured out yet why it was still showing the other blocks.